contracts/core/UniversalChannelHandler.sol

```diff
@@ -82,12 +83,38 @@ contract UniversalChannelHandler is IbcReceiverBaseUpgradeable, UUPSUpgradeable,
         bytes32 destPortAddr,
         bytes calldata appData,
         uint64 timeoutTimestamp
-    ) external {
+    ) external returns (uint64 sequence) {
         bytes memory packetData = IbcUtils.toUniversalPacketBytes(
             UniversalPacket(IbcUtils.toBytes32(msg.sender), MW_ID, destPortAddr, appData)
         );
         emit UCHPacketSent(msg.sender, destPortAddr);
-        dispatcher.sendPacket(channelId, packetData, timeoutTimestamp);
+        sequence = dispatcher.sendPacket(channelId, packetData, timeoutTimestamp);
+    }
+
+    /**
+     * @notice Sends a universal packet over an IBC channel
+     * @param channelId The channel ID through which the packet is sent on the dispatcher
+     * @param destPortAddr The destination port address
+     * @param appData The packet data to be sent
+     * @param timeoutTimestamp of when the packet can timeout
+     */
+    function sendUniversalPacketWithFee(
+        bytes32 channelId,
+        bytes32 destPortAddr,
+        bytes calldata appData,
+        uint64 timeoutTimestamp,
+        uint256[2] calldata gasLimits,
+        uint256[2] calldata gasPrices
+    ) external payable returns (uint64 sequence) {
+        // Cache dispatcher for gas savings
+        IbcDispatcher _dispatcher = dispatcher;
```

Comment:
> should we do this with sendUniversalPacket then?

by "this" you mean caching for gas savings yes? 

if so,  we don't do this with `sendUniversalPacket` because there is only one storage read, whereas with `sendUniversalPacketWithFee` there (would have been) two potential storage reads to find the dispatcher address, one to send the packet and the second to query the feeVault. 

Storage reads are more expensive than a memory read + write, so it becomes worth it to cache the dispatcher address in memory when we have more than one storage read. But if we only have one storage read (as in the case of `sendUniversalPacket`, caching is more expensive than just doing the storage read (i.e. it's one storage read vs 1 storage read + 1 memory write)